Boost 1.65.0 build failure on macOS 10.12

osquery 2.7.0

https://jenkins.brew.sh/job/Homebrew%20Core%20Pull%20Requests/7149/version=sierra/console

(duplicate since that will eventually be deleted https://gist.github.com/ilovezfs/3a6cd48e020218cd81275bd55275a7d4)

```
[ 68%] Building CXX object osquery/tables/CMakeFiles/osquery_system_tables.dir/system/darwin/ad_config.cpp.o
cd /tmp/osquery-20170823-4176-tj8qj4/osquery/tables && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DBOOST_NETWORK_ENABLE_HTTPS -DDARWIN -DDARWIN_10_12 -DNDEBUG -DNOMINMAX -DOSQUERY_BUILD_DISTRO=10.12 -DOSQUERY_BUILD_PLATFORM=darwin -DOSQUERY_BUILD_SDK_VERSION=2.7.0 -DOSQUERY_THRIFT="" -DOSQUERY_THRIFT_LIB=thrift -DOSQUERY_THRIFT_POINTER=boost -DOSQUERY_THRIFT_SERVER_LIB=thrift/server -DSTRIP_FLAG_HELP=1 -DTHRIFT_SQUELCH_CONSOLE_OUTPUT=1 -isystem /tmp/osquery-20170823-4176-tj8qj4/third-party/sysroots/darwin -isystem /usr/local/osquery/legacy/include -isystem /usr/local/osquery/include -I/usr/local/osquery/include/openssl -I/tmp/osquery-20170823-4176-tj8qj4/third-party/sqlite3 -I/tmp/osquery-20170823-4176-tj8qj4/include -I/tmp/osquery-20170823-4176-tj8qj4 -I/tmp/osquery-20170823-4176-tj8qj4/generated/gen-cpp  -DNDEBUG -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/libxml2 -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/aws-sdk-cpp/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/cpp-netlib/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/linenoise/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/thrift/include -Wl,-L/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/linenoise/lib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.12    -std=c++14 -stdlib=libc++ -mmacosx-version-min=10.12 -Qunused-arguments -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-local-typedef -Wno-deprecated-register -Wno-unknown-warning-option -Wnon-virtual-dtor -Wchar-subscripts -Wpointer-arith -Woverloaded-virtual -Wformat -Wformat-security -Werror=format-security -Wabi-tag -fpermissive -fstack-protector-all -pipe -fdata-sections -ffunction-sections -Werror=shadow -fvisibility=hidden -fvisibility-inlines-hidden -Os -fPIE -fpie -fPIC -fpic -flto=thin -march=x86-64 -mno-avx -Wall -Wextra -pedantic -Wuseless-cast -Wno-c99-extensions -Wno-zero-length-array -Wno-unused-parameter -Wno-gnu-case-range -Weffc++ -Wshadow -o CMakeFiles/osquery_system_tables.dir/system/darwin/ad_config.cpp.o -c /tmp/osquery-20170823-4176-tj8qj4/osquery/tables/system/darwin/ad_config.cpp
[ 68%] Building CXX object osquery/tables/CMakeFiles/osquery_system_tables.dir/system/darwin/apps.mm.o
cd /tmp/osquery-20170823-4176-tj8qj4/osquery/tables && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DBOOST_NETWORK_ENABLE_HTTPS -DDARWIN -DDARWIN_10_12 -DNDEBUG -DNOMINMAX -DOSQUERY_BUILD_DISTRO=10.12 -DOSQUERY_BUILD_PLATFORM=darwin -DOSQUERY_BUILD_SDK_VERSION=2.7.0 -DOSQUERY_THRIFT="" -DOSQUERY_THRIFT_LIB=thrift -DOSQUERY_THRIFT_POINTER=boost -DOSQUERY_THRIFT_SERVER_LIB=thrift/server -DSTRIP_FLAG_HELP=1 -DTHRIFT_SQUELCH_CONSOLE_OUTPUT=1 -isystem /tmp/osquery-20170823-4176-tj8qj4/third-party/sysroots/darwin -isystem /usr/local/osquery/legacy/include -isystem /usr/local/osquery/include -I/usr/local/osquery/include/openssl -I/tmp/osquery-20170823-4176-tj8qj4/third-party/sqlite3 -I/tmp/osquery-20170823-4176-tj8qj4/include -I/tmp/osquery-20170823-4176-tj8qj4 -I/tmp/osquery-20170823-4176-tj8qj4/generated/gen-cpp  -DNDEBUG -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/libxml2 -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/aws-sdk-cpp/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/cpp-netlib/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/linenoise/include -I/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/thrift/include -Wl,-L/private/tmp/osquery-20170823-4176-tj8qj4/brew_vendor/linenoise/lib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.12    -std=c++14 -stdlib=libc++ -mmacosx-version-min=10.12 -Qunused-arguments -Wstrict-aliasing -Wno-missing-field-initializers -Wno-unused-local-typedef -Wno-deprecated-register -Wno-unknown-warning-option -Wnon-virtual-dtor -Wchar-subscripts -Wpointer-arith -Woverloaded-virtual -Wformat -Wformat-security -Werror=format-security -Wabi-tag -fpermissive -fstack-protector-all -pipe -fdata-sections -ffunction-sections -Werror=shadow -fvisibility=hidden -fvisibility-inlines-hidden -Os -fPIE -fpie -fPIC -fpic -flto=thin -march=x86-64 -mno-avx -Wall -Wextra -pedantic -Wuseless-cast -Wno-c99-extensions -Wno-zero-length-array -Wno-unused-parameter -Wno-gnu-case-range -Weffc++ -Wshadow -x objective-c++ -fobjc-arc -Wno-c++11-extensions -o CMakeFiles/osquery_system_tables.dir/system/darwin/apps.mm.o -c /tmp/osquery-20170823-4176-tj8qj4/osquery/tables/system/darwin/apps.mm
In file included from /tmp/osquery-20170823-4176-tj8qj4/osquery/tables/system/darwin/apps.mm:19:
In file included from /tmp/osquery-20170823-4176-tj8qj4/include/osquery/core.h:19:
In file included from /usr/local/include/boost/thread/shared_mutex.hpp:22:
In file included from /usr/local/include/boost/thread/pthread/shared_mutex.hpp:13:
In file included from /usr/local/include/boost/thread/mutex.hpp:16:
In file included from /usr/local/include/boost/thread/pthread/mutex.hpp:23:
/usr/local/include/boost/thread/pthread/pthread_mutex_scoped_lock.hpp:33:26: error: expected member name or ';' after declaration specifiers
            void check() BOOST_NOEXCEPT
            ~~~~         ^
/usr/local/include/boost/config/detail/suffix.hpp:959:26: note: expanded from macro 'BOOST_NOEXCEPT'
#  define BOOST_NOEXCEPT noexcept
                         ^
In file included from /tmp/osquery-20170823-4176-tj8qj4/osquery/tables/system/darwin/apps.mm:19:
In file included from /tmp/osquery-20170823-4176-tj8qj4/include/osquery/core.h:19:
In file included from /usr/local/include/boost/thread/shared_mutex.hpp:22:
In file included from /usr/local/include/boost/thread/pthread/shared_mutex.hpp:14:
In file included from /usr/local/include/boost/thread/condition_variable.hpp:16:
In file included from /usr/local/include/boost/thread/pthread/condition_variable.hpp:12:
/usr/local/include/boost/thread/pthread/thread_data.hpp:213:13: error: expected member name or ';' after declaration specifiers
            {
            ^
In file included from /tmp/osquery-20170823-4176-tj8qj4/osquery/tables/system/darwin/apps.mm:19:
In file included from /tmp/osquery-20170823-4176-tj8qj4/include/osquery/core.h:19:
In file included from /usr/local/include/boost/thread/shared_mutex.hpp:22:
In file included from /usr/local/include/boost/thread/pthread/shared_mutex.hpp:14:
In file included from /usr/local/include/boost/thread/condition_variable.hpp:16:
/usr/local/include/boost/thread/pthread/condition_variable.hpp:82:43: error: expected unqualified-id
            check_for_interruption.check();
                                          ^
/usr/local/include/boost/thread/pthread/condition_variable.hpp:116:43: error: expected unqualified-id
            check_for_interruption.check();
                                          ^
/usr/local/include/boost/thread/pthread/condition_variable.hpp:193:47: error: expected unqualified-id
                check_for_interruption.check();
                                              ^
/usr/local/include/boost/thread/pthread/condition_variable.hpp:423:45: error: expected unqualified-id
              check_for_interruption.check();
                                            ^
6 errors generated.
make[2]: *** [osquery/tables/CMakeFiles/osquery_system_tables.dir/system/darwin/apps.mm.o] Error 1
make[1]: *** [osquery/tables/CMakeFiles/osquery_system_tables.dir/all] Error 2
make: *** [all] Error 2
```

CC @theopolis 

Cf. https://github.com/Homebrew/homebrew-core/pull/17150.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -443,6 +443,9 @@ add_definitions(
   -DSTRIP_FLAG_HELP=1
   -DBOOST_NETWORK_ENABLE_HTTPS
   -DNOMINMAX
+
+  # See boost 1.65 breakage https://github.com/Homebrew/homebrew-core/pull/17150
+  -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0
 )
 
 # Remove console output (not handled by logged).

